velox/type/DecimalUtil.h

```diff
@@ -271,6 +271,37 @@ class DecimalUtil {
     return overflow;
   }
 
+  /// This method is used to correct the sum result calculated using
+  /// addWithOverflow. In the addWithOverflow method, when summing two int128_t
+  /// with the same sign, it first converts them to uint128_t for calculation.
+  /// The returned sum represents the lower 127 bits of the uint128_t sum.
+  /// The overflow variable indicates whether an overflow occurred during the
+  /// calculation. If overflow = 1, it means there was an upward overflow, and
+  /// if overflow = -1, there was a downward overflow. However, since the sum
+  /// only retains the lower 127 bits, it may miss one calculation of
+  /// +(1 << 127) or -(1 << 127).
+  /// Therefore, we need to make the following adjustments:
```

Comment:
The adjustments are:

1. If overflow = 1 && sum < 0, the calculation missed +(1 << 127). Add 1 << 127 to the sum.
2. If overflow = -1 && sum > 0, the calculation missed -(1 << 127). Subtract 1 << 127 to the sum.